### DISCUSSION TITLE
Fail to get shard Leaders from querycord node=45: node offline channel not avaiable

### DISCUSSION THREAD

**Discussion by evenzzu163** [milvus-io/milvus#33571] [Q&A and General discussion]
Fail to get shard Leaders from querycord node=45: node offline channel not avaiable   这个是什么意思呢。 之前扩容前检索Milvus耗时10s左右后报这个错 ，检查queryNode内存占满了。今天queryNode内存扩容后，业务检索后还是有这个报错， 是不是需要重新创建索引呢？有什么办法解决吗？
目前我们数据量在300w左右 2048维度，IVF_FLAT索引。扩容前两个queryNode节点，每个20g，扩容后每个30g

**Comment by evenzzu163** [comment]
![Uploading 1111.jpg…]()

**Comment by xiaofan-luan** [comment]
尝试重启一下所有的querynode呢？

**Comment by evenzzu163** [comment]
昨天扩容完后刚重启的

---- Replied Message ----
| From | ***@***.***> |
| Date | 06/04/2024 11:57 |
| To | milvus-io/milvus ***@***.***> |
| Cc | evenzzu163 ***@***.***>,
Author ***@***.***> |
| Subject | Re: [milvus-io/milvus] Fail to get shard Leaders from querycord node=45: node offline channel not avaiable (Discussion #33571) |

尝试重启一下所有的querynode呢？

**Comment by evenzzu163** [comment]
![k8sWatch](https://github.com/milvus-io/milvus/assets/171309236/b69f27c2-5c25-4847-b1b9-9881745849d1)
![queryCoordLog](https://github.com/milvus-io/milvus/assets/171309236/fbeb85a1-1266-4ef4-adce-0f99d5e28140)
![queryNode1](https://github.com/milvus-io/milvus/assets/171309236/74724f1d-1910-446f-ba95-8dbb5ab0170e)
![queryNode2](https://github.com/milvus-io/milvus/assets/171309236/9955b8a9-b5f0-4729-9426-a96c925e36c6)
![searchResponse](https://github.com/milvus-io/milvus/assets/171309236/b37f8bcd-b7c7-424f-b7b0-b427b279aad4)
补充描述：k8s helm部署，1个queryCoord节点+2个queryNode节点，主要问题检索时间太长，后报错

**Comment by evenzzu163** [comment]
Milvus是2.3.1版本

**Reply by yhmo** [reply]
这种报错一般是重启能解决。你的截图里并没有什么有效信息。2.3.1版本比较老，升级版本也是一个解决途径。

**Comment by evenzzu163** [comment]
重启哪个节点呢 我们扩容后都会重启的呀

---- Replied Message ----
| From | ***@***.***> |
| Date | 06/05/2024 17:04 |
| To | milvus-io/milvus ***@***.***> |
| Cc | evenzzu163 ***@***.***>,
Author ***@***.***> |
| Subject | Re: [milvus-io/milvus] Fail to get shard Leaders from querycord node=45: node offline channel not avaiable (Discussion #33571) |

这种报错一般是重启能解决。你的截图里并没有什么有效信息。2.3.1版本比较老，升级版本也是一个解决途径。

**Reply by yhmo** [reply]
如果重启不能解决，那么在querycoord或者querynode的日志里应该会有相关错误信息，但你的截图并没有。

**Reply by yhmo** [reply]
升级到2.3的最新一个版本2.3.17是比较好的做法。有些老版本的bug可能现在已经没人记得怎么回事了。

**Comment by evenzzu163** [comment]
有图片的呀 第二张是queryCoord，第三四张是两个queryNode节点，第五张是milvus返回给业务的

**Comment by evenzzu163** [comment]
有什么升级方案嘛，不丢失数据的离线升级

**Reply by yhmo** [reply]
一般直接升级即可：
https://milvus.io/docs/v2.3.x/upgrade_milvus_cluster-operator.md
https://milvus.io/docs/v2.3.x/upgrade_milvus_cluster-helm.md

**Comment by evenzzu163** [comment]
![queryCoordLog](https://github.com/milvus-io/milvus/assets/171309236/b853cad1-b853-49d6-832b-ada23279998e)
![queryNode1](https://github.com/milvus-io/milvus/assets/171309236/5a886fbf-eb69-4ac2-a62d-1acf6bbddca6)
![queryNode2](https://github.com/milvus-io/milvus/assets/171309236/6215e2e2-5868-4ea2-a82b-51a12d9260d3)
能帮忙看下queryCoord  queryNode这几个节点报错原因嘛

**Reply by yhmo** [reply]
截图里只有INFO，没有看到有任何的WARN或者ERROR。如果你只是截图，恐怕截100张图也没用

**Comment by xiaofan-luan** [comment]
请使用exportlog.sh,获取这个集群的日志。
另外，从你的日志显示，你的每个node是有9个多g内存，并没有你说的20g或者30g

**Comment by evenzzu163** [comment]
是的 已经把milvus下节点重启了 目前正常

---- Replied Message ----
| From | ***@***.***> |
| Date | 06/09/2024 23:12 |
| To | milvus-io/milvus ***@***.***> |
| Cc | evenzzu163 ***@***.***>,
Author ***@***.***> |
| Subject | Re: [milvus-io/milvus] Fail to get shard Leaders from querycord node=45: node offline channel not avaiable (Discussion #33571) |

请使用exportlog.sh,获取这个集群的日志。
另外，从你的日志显示，你的每个node是有9个多g内存，并没有你说的20g或者30g

**Reply by xiaofan-luan** [reply]
目前已经恢复了么，赞👍
如果还有类似问题，可以联系我的邮箱

**Comment by evenzzu163** [comment]
好的

---- Replied Message ----
| From | ***@***.***> |
| Date | 06/09/2024 23:25 |
| To | milvus-io/milvus ***@***.***> |
| Cc | evenzzu163 ***@***.***>,
Author ***@***.***> |
| Subject | Re: [milvus-io/milvus] Fail to get shard Leaders from querycord node=45: node offline channel not avaiable (Discussion #33571) |

目前已经恢复了么，赞👍
如果还有类似问题，可以联系我的邮箱

**Comment by evenzzu163** [comment]
感谢

---- Replied Message ----
| From | ***@***.***> |
| Date | 06/09/2024 23:25 |
| To | milvus-io/milvus ***@***.***> |
| Cc | evenzzu163 ***@***.***>,
Author ***@***.***> |
| Subject | Re: [milvus-io/milvus] Fail to get shard Leaders from querycord node=45: node offline channel not avaiable (Discussion #33571) |

目前已经恢复了么，赞👍
如果还有类似问题，可以联系我的邮箱
